### PR TITLE
improve(relayer): Support keepalives over websockets

### PR DESCRIPTION
**Pull request by pxrl**
As newer chains are added, we sometimes see that websockets are occasionally dropped; potentially as a result of low activity on the chain meaning that the websocket sits idle. This commit adds a simple periodic keepalive that will be initiated by each listener process towards each of its configured providers.

### CODE REVIEW

File: src/libexec/RelayerSpokePoolIndexer.ts [hunk 1]
```diff
@@ -291,9 +291,53 @@ async function run(argv: string[]): Promise<void> {
...
-    provider._websocket.on("error", (err) => {
-      const _provider = getOriginFromURL(provider.connection.url);
+    const { _websocket: ws } = provider;
+    const _provider = getOriginFromURL(provider.connection.url);
+    let interval: NodeJS.Timer | undefined;
+    let timeout: NodeJS.Timeout | undefined;
+
+    const cleanup = () => {
+      if (interval) {
+        clearInterval(interval);
```
Comment (nicholaspai) [inline src/libexec/RelayerSpokePoolIndexer.ts:306]: where is `clearInterval/clearTimeout` defined?

**Review by nicholaspai — APPROVED** [review]
Just one 1

File: src/libexec/RelayerSpokePoolIndexer.ts (hunk 1 above)
Comment (pxrl) [reply src/libexec/RelayerSpokePoolIndexer.ts:306]: These are built-ins that are packaged with node: https://nodejs.org/api/timers.html#clearintervaltimeout

File: src/libexec/RelayerSpokePoolIndexer.ts [hunk 2]
```diff
@@ -199,7 +201,7 @@ async function listen(
   });
 
   do {
-    await setTimeout(INDEXER_POLLING_PERIOD);
+    await delay(INDEXER_POLLING_PERIOD);
```
Comment (james-a-morris) [inline src/libexec/RelayerSpokePoolIndexer.ts:204]: OOC can we re-use the SDK's delay function or is this separate?

**Review by james-a-morris — COMMENTED** [review]
one question otherwise this lgtm.

Is there a good/easy way to test these changes?

File: src/libexec/RelayerSpokePoolIndexer.ts (hunk 2 above)
Comment (pxrl) [reply src/libexec/RelayerSpokePoolIndexer.ts:204]: We can use it :+1:

Comment (pxrl) [reply src/libexec/RelayerSpokePoolIndexer.ts]: https://github.com/across-protocol/relayer/pull/1771/commits/be3a943cce89d9f91c9f22a66f41c48f28cd7a53

**Pull request merged**